From 99e51f3bed99e61140959a7f5759221345e61bda Mon Sep 17 00:00:00 2001
From: Bin Yang <bin.yang@intel.com>
Date: Tue, 18 Sep 2018 01:39:51 +0000
Subject: [PATCH] fix cdev state error while trip point is off
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

When the trip point is off, it will call thd_cdev_set_state(state=0)
to update cdev state.

if zone_trip_limits.size() == 0, it means all trip points using this cdev
are off. It needs to set this cdev to min state (target_state_valid = 0).

Signed-off-by: Bin Yang <bin.yang@intel.com>
---
 src/thd_cdev.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/thd_cdev.cpp b/src/thd_cdev.cpp
index 52ae3af..094fb27 100644
--- a/src/thd_cdev.cpp
+++ b/src/thd_cdev.cpp
@@ -258,6 +258,8 @@ int cthd_cdev::thd_cdev_set_state(int set_point, int target_temp,
 				if (_target_state_valid)
 					target_value = get_min_state();
 			}
+		} else {
+			target_state_valid = 0;
 		}
 	}
 
-- 
2.19.1

